bmc/redfish.go

```diff
@@ -36,6 +46,8 @@ func NewRedfishBMCClient(
 	ctx context.Context,
 	endpoint, username, password string,
 	basicAuth bool,
+
+	options PollingOptions,
```

Comment:
Superfluos newline. Given the amount of optiones, it may be worth introducing a `BMCClientOptions` struct to hold all of them.

Reply:
I had that already implemented in the las commit (https://github.com/ironcore-dev/metal-operator/pull/174/commits/c7bf55023862dfad777d8fc1f49019b6e6d30b59). 
Some of the settings are mandatory like user/pw, so I wanted to separate them